pylint/reporters/text.py

```diff
@@ -64,64 +79,46 @@
 }
 
 
-def _get_ansi_code(color=None, style=None):
+def _get_ansi_code(msg_style: MessageStyle) -> str:
     """return ansi escape code corresponding to color and style
 
-    :type color: str or None
-    :param color:
-      the color name (see `ANSI_COLORS` for available values)
-      or the color number when 256 colors are available
-
-    :type style: str or None
-    :param style:
-      style string (see `ANSI_COLORS` for available values). To get
-      several style effects at the same time, use a coma as separator.
+    :param msg_style: the message style
 
     :raise KeyError: if an unexistent color or style identifier is given
 
-    :rtype: str
     :return: the built escape code
     """
     ansi_code = []
-    if style:
-        style_attrs = utils._splitstrip(style)
-        for effect in style_attrs:
+    if msg_style.style == ():
+        for effect in msg_style.style:
             ansi_code.append(ANSI_STYLES[effect])
-    if color:
-        if color.isdigit():
+    if msg_style.colour:
+        if msg_style.colour.isdigit():
             ansi_code.extend(["38", "5"])
-            ansi_code.append(color)
+            ansi_code.append(msg_style.colour)
         else:
-            ansi_code.append(ANSI_COLORS[color])
+            ansi_code.append(ANSI_COLORS[msg_style.colour])
     if ansi_code:
         return ANSI_PREFIX + ";".join(ansi_code) + ANSI_END
     return ""
 
 
-def colorize_ansi(msg, color=None, style=None):
+def colorize_ansi(msg: str, msg_style: MessageStyle) -> str:
```

Comment:
This will break backwards compatibility. We might have to use some tricks, but what it comes down to: For the transition phase, users should still be able to use the old function signature.

Only a rough idea:
```py
def colorize_ansi(
    msg: str,
    msg_style: Union[MessageStyle, str, None] = None,  # needs to accept an old 'color' argument
    style: Optional[str] = None,
    **kwargs: Optional[str],  # if 'color' is passed as keyword argument
) -> str:
    if not isinstance(msg_style, MessageStyle):
        warnings.warn("...", DeprecationWarning)
        color = kwargs.get("color")
        style_attrs = tuple(utils._splitstrip(style))
        msg_style = MessageStyle(color or msg_style, style_attrs)
    ...
```